### PR TITLE
Remove Makefile

### PR DESCRIPTION
**Pull request by leezu**
Replaced by cmake build. See https://github.com/apache/incubator-mxnet/issues/16167

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,768 +0,0 @@
...
-test: $(TEST)
-
-lint: cpplint pylint
-
-cpplint:
-	3rdparty/dmlc-core/scripts/lint.py mxnet cpp include src plugin tests \
-	--exclude_path src/operator/contrib/ctc_include include/mkldnn
-
-pylint:
-	python3 -m pylint --rcfile=$(ROOTDIR)/ci/other/pylintrc --ignore-patterns=".*\.so$$,.*\.dll$$,.*\.dylib$$" python/mxnet
```
Comment (szha) [inline Makefile:676]: should we add these somewhere in the cmake?

Comment (szha) [reply Makefile:676]: better yet if we could have them as a git hook before check-in

Comment (leezu) [reply Makefile:676]: Developers may run `python ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh sanity_check` to reproduce the CI Sanity check. We can provide git hooks in a separate effort

**Comment by leezu** [comment]
@mxnet-bot run ci [centos-cpu]

**Comment by leezu** [comment]
@mxnet-bot run ci [all]

**Comment by szha** [comment]
Developers may still need to rely on some of the Makefile targets for development purpose. Make sure that they are documented prominently, as we can't ask developers to read the CI code first when starting with out project.

**Comment by leezu** [comment]
The Makefile targets are undocumented and are tied to the CI:

https://github.com/apache/incubator-mxnet/blob/faccd91071cc34ed0b3a192d3c7932441fe7e35e/ci/docker/install/requirements#L35-L36

For now using the `python ci/build.py --platform ubuntu_cpu /work/runtime_functions.sh sanity_check` or manually running `pylint` and `cpplint` should be acceptable. We can have a separate PR

**Comment by szha** [comment]
Thanks for elaborating. Would you update the document here to reflect this? https://cwiki.apache.org/confluence/display/MXNET/Development+Process

**Comment by leezu** [comment]
I updated the https://cwiki.apache.org/confluence/display/MXNET/Development+Process and 
https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=125309983&src=contextnavpagetreemode
Searching through the wiki didn't yield other obvious references to the Makefile

**Pull request merged**